src/pybind/mgr/volumes/fs/async_job.py

```diff
@@ -165,10 +197,33 @@ def run(self):
                     # Increase concurrency: create more threads.
                     log.debug("creating new threads to job increase")
                     for i in range(c, self.nr_concurrent_jobs):
-                        self.threads.append(JobThread(self, self.vc, name="{0}.{1}.{2}".format(self.name_pfx, time.time(), i)))
-                        self.threads[-1].start()
+                        if not self.run_event.is_set():
```

Comment:
I don't understand this? Why is this required and more so why is it breaking out of the thread?

Reply:
So that no new threads are spawned when entire async op machinery has been paused.

Reply:
As soon as async op machinery is resumed by the user, the jobs will be continue to run and new thread will also be spawned.

Reply:
Marking this comment resolved since it was discussed and changes have moved forward since.